WordPress/openverse-catalog
# Remove legacy provider DAG logic

## Fixes


Fixes #824 by @AetherUnbound 
## Description


* Removes now-unused logic branches that handled legacy provider scripts
* Updates the ProviderWorkflow and ProviderReingestionWorkflow dataclasses to infer the provider script, provider name, and media types from the configured ingestion class
  * This means that to configure a new DAG, *all* you need to define in the dataclass is the Ingester class! Everything else is optional
* Updates some docs

**Note: #848 should be merged first and this will need to be rebased to pull in Europeana changes.**

Not included:
It may be time to revisit the `generate_tsv_filenames` logic, which I have not touched in this PR beyond deleting fully-unused legacy branches. Should we open a new issue for this?

## Testing Instructions

```just test```

Test some DAGs! Make sure to try at least one audio-only, one image-only, one audio-and-image, and one reingestion workflow. If you're looking for a thrill set `INGESTION_LIMIT` to 10 and turn them all on 🥳 

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
